### PR TITLE
Enable NRT in AbstractSyncNamespaceCodeRefactoringProvider

### PR DESCRIPTION
**Pull request by genlu**
Fix https://github.com/dotnet/roslyn/issues/60270

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -28,14 +26,15 @@ public CSharpSyncNamespaceCodeRefactoringProvider()
...
         }
 
-        protected override async Task<SyntaxNode> TryGetApplicableInvocationNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        protected override async Task<SyntaxNode?> TryGetApplicableInvocationNodeAsync(Document document, TextSpan span, CancellationToken cancellationToken)
         {
             if (!span.IsEmpty)
                 return null;
 
-            var position = span.Start;
+            if (await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false) is not CompilationUnitSyntax compilationUnit)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs:34]: if we're exported for LangNamesd.CSharp, this should always work, right?  can you se GetRequiredSyntaxRootASync?

File: src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs [hunk 2]
```diff
@@ -152,36 +150,40 @@ internal sealed class State
             /// </summary>
             private static bool IsDocumentPathRootedInProjectFolder(Document document)
             {
+                var absoluteDircetoryPath = PathUtilities.GetDirectoryName(document.FilePath);
```
[the remark this answers is not in the document]
Comment (genlu) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs:153]: Ah, I missed that. Will fix

[the remark this answers is not in the document]
Comment (KirillOsenkov) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs:153]: @genlu I've filed https://github.com/dotnet/roslyn/issues/62507 to track this :)

[the remark this answers is not in the document]
Comment (genlu) [reply src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs:153]: Thanks!

**Pull request merged**